src/api/modules/dispatch/allowance.ts

```diff
@@ -23,24 +23,19 @@ export async function ensureAllowanceDispatch(
   const arweave = new Arweave(defaultGateway);
 
   // allowance or sign auth
-  try {
-    if (alwaysAsk) {
-      const address = await arweave.wallets.jwkToAddress(keyfile);
 
-      await signAuth(
-        appData,
-        // @ts-expect-error
-        dataEntry.toJSON(),
-        address,
-      );
-    }
+  if (alwaysAsk) {
+    const address = await arweave.wallets.jwkToAddress(keyfile);
 
-    // if (allowance.enabled) {
-    //   await allowanceAuth(appData, allowance, price, alwaysAsk);
-    // }
-  } catch (e) {
-    freeDecryptedWallet(keyfile);
```

Comment:
`dispatch.background.ts` calls `ensureAllowanceDispatch()` a second time if the first upload attempt fails, so if we call `freeDecryptedWallet()` here, the second one will fail too because the keyfile will no longer be there.